### PR TITLE
Bridge igmp group version

### PR DESCRIPTION
**Pull request by ssahani**
```
ip -d link show bridge99                                                      ✔  10094  18:54:01 
11: bridge99: <NO-CARRIER,BROADCAST,MULTICAST,UP> mtu 1500 qdisc noqueue state DOWN mode DEFAULT group default qlen 1000
    link/ether e2:11:4e:b3:8b:64 brd ff:ff:ff:ff:ff:ff promiscuity 0 minmtu 68 maxmtu 65535 
    bridge forward_delay 900 hello_time 900 max_age 900 ageing_time 900 stp_state 1 priority 9 vlan_filtering 0 vlan_protocol 802.1Q bridge_id 0009.0:0:0:0:0:0 designated_root 0009.0:0:0:0:0:0 root_port 0 root_path_cost 0 topology_change 0 topology_change_detected 0 hello_timer    4.27 tcn_timer    0.00 topology_change_timer    0.00 gc_timer    4.25 vlan_default_pvid 1 vlan_stats_enabled 0 group_fwd_mask 0 group_address 01:80:c2:00:00:00 mcast_snooping 1 mcast_router 1 mcast_query_use_ifaddr 0 mcast_querier 1 mcast_hash_elasticity 16 mcast_hash_max 4096 mcast_last_member_count 2 mcast_startup_query_count 2 mcast_last_member_interval 100 mcast_membership_interval 26000 mcast_querier_interval 25500 mcast_query_interval 12500 mcast_query_response_interval 1000 mcast_startup_query_interval 3125 mcast_stats_enabled 0 mcast_igmp_version 3 mcast_mld_version 1 nf_call_iptables 0 nf_call_ip6tables 0 nf_call_arptables 0 addrgenmode eui64 numtxqueues 1 numrxqueues 1 gso_max_size 65536 gso_max_segs 65535 

````

```
cat 25-bridge.netdev                                                                                            ✔  ⚡  963  18:57:07 
[NetDev]
Name=bridge99
Kind=bridge

[Bridge]
HelloTimeSec=9
MaxAgeSec=9
ForwardDelaySec=9
AgeingTimeSec=9
Priority=9
MulticastQuerier= true
MulticastSnooping=true
STP=true
IGMPVersion=3

```

### CODE REVIEW

File: man/systemd.netdev.xml [hunk 1]
```diff
@@ -412,6 +412,14 @@
           </para>
         </listitem>
       </varlistentry>
+      <varlistentry>
+        <term><varname>IGMPVersion=</varname></term>
+        <listitem>
+          <para>Allows to change bridge's multicast cast Internet Group Management Protocol (IGMP) version.
```
Comment (yuwata) [inline man/systemd.netdev.xml]: `multicast cast`??

File: src/network/netdev/bridge.c [hunk 2]
```diff
@@ -133,6 +133,12 @@ static int netdev_bridge_post_create(NetDev *netdev, Link *link, sd_netlink_mess
                         return log_netdev_error_errno(netdev, r, "Could not append IFLA_BR_STP_STATE attribute: %m");
         }
 
+        if (IN_SET(b->igmp_version, 2,3)) {
```
Comment (yuwata) [inline src/network/netdev/bridge.c]: Maybe, we should introduce `bridge_verify()` to check the value.

File: man/systemd.netdev.xml (hunk 1 above)
Comment (ssahani) [reply man/systemd.netdev.xml]: ugg !

File: src/network/netdev/bridge.c (hunk 2 above)
Comment (ssahani) [reply src/network/netdev/bridge.c]: yes that would be better in the next version.  good idea.

**Comment by ssahani** [comment]
updated thanks for the review @yuwata .

File: src/network/netdev/bridge.c (hunk 2 above)
Comment (poettering) [reply src/network/netdev/bridge.c]: i'd probably add a specific parser just for this

Comment (ssahani) [reply src/network/netdev/bridge.c]: OK that would also do thanks.

**Comment by ssahani** [comment]
updated thanks for review @poettering

File: src/network/netdev/bridge.c [hunk 3]
```diff
@@ -287,6 +293,36 @@ int link_set_bridge(Link *link) {
...
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+        assert(data);
+
+        r = safe_atou8(rvalue, &u);
+        if (r < 0 || !IN_SET(u, 2, 3)) {
+                log_syntax(unit, LOG_ERR, filename, line, r, "bridge IGMP version number '%s' is invalid, ignoring assignment: %s", lvalue, rvalue);
```
Comment (yuwata) [inline src/network/netdev/bridge.c]: Inserting `lvalue` sounds strange for me.
How about the following:
`"Invalid bridge IGMP version number, ignoring assignment: %s", rvalue`
or
`"Invalid bridge IGMP version number '%s', ignoring assignment.", rvalue`

Comment (ssahani) [reply src/network/netdev/bridge.c]: updated thanks for the comment.

**Comment by yuwata** [comment]
I've added a test for the setting. Setting the green label.

**Comment by yuwata** [comment]
CI failures are unrelated.

**Pull request merged**